dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java

```diff
@@ -85,6 +104,35 @@ private String toJsonAsString(final AppSecrets object) throws DotDataException {
         }
     }
 
+    /**
+     * This method takes a byte array and converts its contents into a char array
+     * No String middle man is created.
+     * https://stackoverflow.com/questions/8881291/why-is-char-preferred-over-string-for-passwords
+     * @param bytes
+     * @return
+     */
+    char[] bytesToCharArrayUTF(byte[] bytes) throws IOException {
+        final BufferedReader reader = new BufferedReader(
+                new InputStreamReader(new ByteArrayInputStream(bytes), StandardCharsets.UTF_8));
+        final List<Integer> integers = new ArrayList<>(bytes.length);
+        int chr;
+        while ((chr = reader.read()) != -1) {
+            integers.add(chr);
+        }
+        return ArrayUtils.toPrimitive(
+                integers.stream().map(value -> (char) value.intValue()).toArray(Character[]::new));
+    }
+
+    /**
+     * This method takes a char array and converts its contents into a byte array No String middle
+     * man is created. https://stackoverflow.com/questions/8881291/why-is-char-preferred-over-string-for-passwords
+     */
+    byte[] charsToBytesUTF(final char[] chars) throws IOException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5329810)